halo/attest/keeper/keeper.go

```diff
@@ -254,6 +256,15 @@ func (k *Keeper) addOne(ctx context.Context, agg *types.AggVote, valSetID uint64
 			)
 		} else if err != nil {
 			return errors.Wrap(err, "insert signature")
+		} else {
+			insertSig = true
+		}
+	}
+
+	if insertAtt && !insertSig {
```

Comment:
Shouldn't we account for early returns in line 230 or 258?

Moreover, should we only revert the attestations and not all the other inserted signatures before one of them generates an error?

Reply:
In general, errors inside cosmos modules are not supported or expected. Most errors returned by keeper logic will result in the chain stalling (or proposals being rejected). So no need to worry about cleanup or ordering in these cases, as the whole block will basically fail.